Document mobile game automation (#3527)
saucelabs/sauce-docs: docs/mobile-apps/automated-testing/alttester/unreal.md

---
id: unreal
title: AltTester for Unreal Engine
sidebar_label: Unreal Engine
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

This page walks you through running AltTester-driven C# tests against an **Unreal Engine** game on Sauce Labs real devices. For background on what AltTester is and how it talks to Sauce, see the [AltTester overview](/mobile-apps/automated-testing/alttester).

## Prerequisites

- An active [Sauce Labs](https://saucelabs.com/) account.
- An Unreal Engine project that you can build for Android or iOS.

> - **AltTester Unreal SDK 1.1 or later** — install as an Unreal plugin in your project.

Reviewer: ```suggestion
- **AltTester Unreal SDK 1.1 or later** — download the AltTester® Plugin from the [AltTester® website](https://alttester.com/downloads/). See the [Get Started guide](https://alttester.com/docs/unreal-sdk/latest/pages/get-started.html) for full instructions..
```

- **AltTester-Driver 2.3 or later** — .NET NuGet package [`AltTester-Driver`](https://www.nuget.org/packages/AltTester-Driver).
- **Appium.WebDriver 8 or later** — .NET NuGet package [`Appium.WebDriver`](https://www.nuget.org/packages/Appium.WebDriver).

> - **AltTester Desktop**, installed locally or on a VM.

Reviewer: ```suggestion
- **AltTester Desktop**, installed locally or on a VM. Can be downloaded from the [AltTester® website](https://alttester.com/downloads/).
```

- **Sauce Connect Proxy** client (recommended path; WebSocket over the tunnel is supported).

## Instrument Your Unreal Build

1. Download the AltTester Unreal SDK from AltTester (see the [References](#references) section below).

> 2. Place the plugin under `<YourProject>/Plugins/AltTester/` in your Unreal project directory.

Reviewer: ```suggestion
1. Download the AltTester Unreal SDK from the [AltTester® website](https://alttester.com/downloads/)
2. Set up the AltTester® Unreal SDK in your project following the [AltTester® documentation](https://alttester.com/docs/unreal-sdk/latest/pages/get-started.html#set-up-the-alttester-unreal-sdk-in-your-project)
```

- **iOS XCUITest behavior differences:** on `appium3-2026-01` and later, the official Appium WebDriverAgent is used. Some endpoints may behave differently than on older images that used SauceWebDriverAgent. See [Appium real-devices](/mobile-apps/automated-testing/appium/real-devices#webdriveragent-for-ios-real-devices) for details.

## References

- [AltTester® for Unreal Engine](https://alttester.com/alttester-unreal/) — AltTester® product page for Unreal.
- [AltTester® documentation](https://alttester.com/docs/) — full SDK and driver reference.
- [Unreal-LyraStarterGame-Tests](https://github.com/alttester/Unreal-LyraStarterGame-Tests) — AltTester®'s public C# test project against the Lyra Starter Game, useful as a copy-from-and-adapt example.
- [Unreal-LyraStarterGame](https://github.com/alttester/Unreal-LyraStarterGame) — the example Unreal project with the AltTester® plugin pre-installed.
- [AltTester-Driver on NuGet](https://www.nuget.org/packages/AltTester-Driver) — current .NET driver versions (works for both engines).

## See Also

> 

Reviewer: ```suggestion

- See the [AltTester® Sauce Labs integration guide](https://alttester.com/docs/sdk/latest/pages/alttester-with-cloud.html#saucelabs) for the full walkthrough including Sauce Connect setup and the complete `BaseTest.cs` example.
```

- **iOS XCUITest behavior differences:** on `appium3-2026-01` and later, the official Appium WebDriverAgent is used. Some endpoints may behave differently than on older images that used SauceWebDriverAgent. See [Appium real-devices](/mobile-apps/automated-testing/appium/real-devices#webdriveragent-for-ios-real-devices) for details.

## References

- [AltTester® for Unreal Engine](https://alttester.com/alttester-unreal/) — AltTester® product page for Unreal.
- [AltTester® documentation](https://alttester.com/docs/) — full SDK and driver reference.
- [Unreal-LyraStarterGame-Tests](https://github.com/alttester/Unreal-LyraStarterGame-Tests) — AltTester®'s public C# test project against the Lyra Starter Game, useful as a copy-from-and-adapt example.
- [Unreal-LyraStarterGame](https://github.com/alttester/Unreal-LyraStarterGame) — the example Unreal project with the AltTester® plugin pre-installed.
- [AltTester-Driver on NuGet](https://www.nuget.org/packages/AltTester-Driver) — current .NET driver versions (works for both engines).

## See Also

> - See the [AltTester® Sauce Labs integration guide](https://alttester.com/docs/sdk/latest/pages/alttester-with-cloud.html#saucelabs) for the full walkthrough including Sauce Connect setup and the complete `BaseTest.cs` example.

Reviewer: ```suggestion
- See the [AltTester® Sauce Labs integration guide](https://alttester.com/docs/unreal-sdk/latest/pages/alttester-with-cloud.html#saucelabs) for the full walkthrough including Sauce Connect setup and the complete `BaseTest.cs` example.
```
